config/public-api-v1.yml

```diff
@@ -31,31 +31,365 @@ paths:
+            innovation in education theory, practice and policy. This programme
+            aims to inform and inspire, to challenge orthodoxies and encourage a
+            freshness of vision. It provides support and guidance for learning
+            and teaching in art and design, identifying strategies to motivate
+            and engage pupils in making, discussing and evaluating visual and
+            material culture. The IOE provides excellent studio space and
+            facilities for Art and Design, including a computing suite where
+            students can learn how technology is used in art and design
+            education. The programme also has strong links with galleries,
+            museums and other sites for learning, which are recognised as an
+            important resource for engaging students in cultural and social
+            issues.Through seminars and studio-based activities, students will
+            study the concepts, processes and skills of art, craft and design,
+            sharing their knowledge and understanding with other student
+            teachers and consider how it relates to the secondary curriculum.
+            Towards the end of the PGCE, students will build on their own
+            practice by initiating a curriculum development project, culminating
+            in the display of their own work in a final exhibition that
+            represents their personal philosophy for art and design education.
+        accredited_body_code:
+          type: string
+          description: |
+            Unique provider-code for the accredited body of this course. Only
+            present if the course is accredited by a provider other than the one
+            running this course.
+          maxLength: 3
+          example: 2FR
+        age_range_in_years:
+          type: string
+          description: |
+            Description of the age-range of the pupils that a course is specified for.
+          example: 11_to_14
+        applications_open_from:
+          type: string
+          format: date
+          description: "Date from which applications can be submitted."
+          example: "2019-10-08"
+        bursary_amount:
+          type: integer
+          description: "Bursary amount for this course."
+          example: 9000
+        bursary_requirements:
+          type: string
+          description: "Description of requirements to be eligible for a bursary."
+          example: "a degree of 2:2 or above in any subject"
+        changed_at:
+          type: string
+          format: date-time
+          description: |
+            Date-time timestamp of when this course or any of it's related data changed.
+          example: "2019-06-13T10:44:31Z"
         code:
           type: string
+          description: |
+            Code that uniquely identifies this course within it's providers list of courses.
+          example: 3GTY
+          maxLength: 4
+        course_length:
```

Comment:
Again, is a string the correct thing here? `course_length_in_months` or `course_length_in_years` maybe?

Reply:
Also a good idea, but this is going to be hard to implement, I think. Here's what a sample of the data looks like:

```ruby
> CourseEnrichment.published.where(course_id: rc.courses.select(:id)).map(&:course_length).uniq
=> ["OneYear",
 "1 Year for Full Time from September to end of June/ 2 years part time (2 days in Year 1 and 3 days in Year 2). Opportunity to train on 5 term training programme part time, completing at Easter of Year 2. ",
 "Up to 4 terms ",
 "TwoYears",
 "10 months",
 "Other",
 "The School Direct salaried PGCE with QTS placement is from August 2020 through to the end of the Summer Term July 2021",
 "9 Months",
 "1 year (3 terms) full-time or 4 terms part-time",
 ...
```

Reply:
Stored as such
`OneYear`
`TwoYears`

but render out as human friendly text

Reply:
We could do it in months then and translate it in the serializer until we're in a position to fix the data.

Reply:
I'm not clear on how we'd translate the existing data.